### PR TITLE
Add early stopping section

### PR DESCRIPTION
**Pull request by brandenchan**
Add a section describing early stopping with code example.

### CODE REVIEW

File: docs/latest/pipeline_nodes/reader.mdx [hunk 1]
```diff
@@ -300,6 +300,35 @@ This can be particularly useful if you want to upload the model to the Hugging F
 transformers_models = reader.inferencer.model.convert_to_transformers()
 ```
 
+Instead of defining a fixed number of training epochs, you can train the model using a method called early stopping.
+This performs cycles of training and evaluation until the model is no longer improving.
+To go for this approach, run:
```
Comment (sjrl) [inline docs/latest/pipeline_nodes/reader.mdx]: I would suggest changing this to `To use this approach, run:`

File: docs/latest/pipeline_nodes/reader.mdx [hunk 2]
```diff
@@ -300,6 +300,35 @@ This can be particularly useful if you want to upload the model to the Hugging F
...
+
+```python
+from haystack.nodes import FARMReader
+from haystack.modeling.training.base import EarlyStopping
```
Comment (sjrl) [inline docs/latest/pipeline_nodes/reader.mdx]: I would update this import to `from haystack.utils.early_stopping import EarlyStopping`

File: docs/latest/pipeline_nodes/reader.mdx [hunk 3]
```diff
@@ -300,6 +300,35 @@ This can be particularly useful if you want to upload the model to the Hugging F
...
+
+early_stopping = EarlyStopping(
+        metric='top_n_accuracy',
```
Comment (sjrl) [inline docs/latest/pipeline_nodes/reader.mdx]: Consider updating the spacing to use 4 spaces to match the formatting of the code used earlier in this doc.

File: docs/latest/pipeline_nodes/reader.mdx [hunk 4]
```diff
@@ -300,6 +300,35 @@ This can be particularly useful if you want to upload the model to the Hugging F
...
+    metric='top_n_accuracy',
+    save_dir='early_stop_model',
+    mode='max',
+    patience=10,
+    min_delta=0.001,
+    min_evals=0,
+)
+reader = FARMReader(model_name_or_path='deepset/roberta-base-squad2', use_gpu=True)
+reader.train(
+        data_dir='./data',
```
Comment (sjrl) [inline docs/latest/pipeline_nodes/reader.mdx]: Update indents here as well please :)

**Review by sjrl — APPROVED** [review]
Looks good! Thanks for adding this :)

File: docs/latest/pipeline_nodes/reader.mdx [hunk 5]
```diff
@@ -300,6 +300,35 @@ This can be particularly useful if you want to upload the model to the Hugging F
 transformers_models = reader.inferencer.model.convert_to_transformers()
 ```
 
+Instead of defining a fixed number of training epochs, you can train the model using a method called early stopping.
+This performs cycles of training and evaluation until the model is no longer improving.
```
Comment (agnieszka-m) [inline docs/latest/pipeline_nodes/reader.mdx]:
```suggestion
This method performs cycles of training and evaluation until the model is no longer improving.
```

**Review by agnieszka-m — CHANGES_REQUESTED** [review]

File: docs/latest/pipeline_nodes/reader.mdx (hunk 5 above)
Comment (sjrl) [reply docs/latest/pipeline_nodes/reader.mdx]: Thanks for the change! And my apologies for approving the commit @brandenchan. I thought this was the PR in Haystack and not for the website.

**Pull request merged**